nomos-sdp/src/lib.rs

```diff
@@ -51,12 +52,71 @@ pub enum ServiceType {
     ExecutorNetwork,
 }
 
+// Add Display and FromStr implementations so we can serialize
+// Hashmap<ServiceType, ...> easily
+impl std::fmt::Display for ServiceType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::BlendNetwork => write!(f, "BN"),
+            Self::DataAvailability => write!(f, "DA"),
+            Self::ExecutorNetwork => write!(f, "EX"),
+        }
+    }
+}
+
+impl FromStr for ServiceType {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "BN" => Ok(Self::BlendNetwork),
+            "DA" => Ok(Self::DataAvailability),
+            "EX" => Ok(Self::ExecutorNetwork),
+            _ => Err(format!("Unknown ServiceType: {s}")),
+        }
+    }
+}
+
 pub type Nonce = [u8; 16];
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
-#[serde(transparent)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
 pub struct ProviderId(pub [u8; 32]);
 
+impl Serialize for ProviderId {
```

Comment:
@bacv , @danielSanchezQ 
The similar question as ServiceType. 

The only difference is that it might be useful for ProviderId to be base64 string, if we are going to pass it to configs.

Reply:
Could we change that to hex strings? I've seen we use `const-hex` in such cases, would be better to have everything in one format.

Reply:
Fixed: https://github.com/logos-co/nomos/pull/1368/commits/522ad16fe107467a430ee4334cf7e045629055a9.